From feff2ec80eeeba5074d19efcddf4867adf639b4f Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Mon, 5 Jul 2010 09:07:08 +1000
Subject: [PATCH] uxa: don't compare planemask with FB_ALLONES.

planemask is an unsigned long initialised to ~0, on 64-bit this is not equal
to an (unsigned int)-1.

Use the macro provided to do this.

Signed-off-by: Dave Airlie <airlied@redhat.com>
---
 uxa/uxa-accel.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/uxa/uxa-accel.c b/uxa/uxa-accel.c
index ded66a1..633b433 100644
--- a/uxa/uxa-accel.c
+++ b/uxa/uxa-accel.c
@@ -81,7 +81,7 @@ uxa_fill_spans(DrawablePtr pDrawable, GCPtr pGC, int n,
 	if (!dst_pixmap)
 		goto fallback;
 
-	if (pGC->alu != GXcopy || pGC->planemask != FB_ALLONES)
+	if (pGC->alu != GXcopy || !UXA_PM_IS_SOLID(pDrawable, pGC->planemask))
 		goto solid;
 
 	format = PictureMatchFormat(screen,
@@ -1048,7 +1048,7 @@ uxa_fill_region_solid(DrawablePtr pDrawable,
 	extents = REGION_EXTENTS(screen, pRegion);
 
 	/* Using GEM, the relocation costs outweigh the advantages of the blitter */
-	if (nbox == 1 || (alu != GXcopy && alu != GXclear) || planemask != FB_ALLONES) {
+	if (nbox == 1 || (alu != GXcopy && alu != GXclear) || !UXA_PM_IS_SOLID(&pixmap->drawable, planemask)) {
 try_solid:
 		if (uxa_screen->info->check_solid &&
 		    !uxa_screen->info->check_solid(&pixmap->drawable, alu, planemask))
-- 
1.7.1

